users/mutations/social.py

```diff
@@ -46,4 +48,50 @@ def mutate(root, info, code):
             if split_response_by_ampersand[0]:
                 access_token_array = split_response_by_ampersand[0].split("=")
                 token = access_token_array[1]
-        return GithubAccessToken(access_token=token)
\ No newline at end of file
+        return GithubAccessToken(access_token=token)
+    
+    
+class TwitterRequestToken(graphene.Mutation):
```

Comment:
I am not sure if we need to write a new mutation to handle twitter social auth. We didn't had to write the same for google and facebook. Can you please double check if this is the only way to do this?

Reply:
Google and facebook provides an sdk that can communicate with their API from the frontend itself. Twitter doesn't allow this, throws a cors error if I try to send an API request from the frontend.

https://twittercommunity.com/t/will-twitter-api-support-cors-headers-soon/28276/2

I would need the access token to login through the socialauth graphql mutation, and that can only be done through requests from the backend. I could maybe add a POST method from the next.js server section, haven't really tested that out but I think it basically follows the same process as creating a new graphql mutation, and in this way we can also allow React (not only next.js) applications to use the mutation in sheliak backend to login through twitter.

If you can double check for me, let me know if there are better ways to implement it.

Reply:
Ok that make sense. Thanks.